pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -150,9 +150,11 @@ type BlockStorageOpts struct {
 
 // NetworkingOpts is used for networking settings
 type NetworkingOpts struct {
-	IPv6SupportDisabled bool   `gcfg:"ipv6-support-disabled"`
-	PublicNetworkName   string `gcfg:"public-network-name"`
-	InternalNetworkName string `gcfg:"internal-network-name"`
+	IPv6SupportDisabled  bool     `gcfg:"ipv6-support-disabled"`
+	PublicNetworkName    []string `gcfg:"public-network-name"`
+	InternalNetworkName  []string `gcfg:"internal-network-name"`
+	publicNetworkNames   map[string]struct{}
```

Comment:
I'd suggest to use https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/util/util.go#L30 rather than adding two extra fields here.

Reply:
You are absolutely right. Such a minor performance improvement does not warrant such an ugly solution.

Fixed!